src/hooks/useActionCenterSubscriber/useTradeActionSubscriber.ts

```diff
@@ -0,0 +1,212 @@
+        actionCenterSlice.actions.upsertAction({
+          type: ActionCenterType.Swap,
+          status: ActionStatus.Pending,
+          title: translate('notificationCenter.notificationTitle', {
+            sellAmountAndSymbol: toCrypto(
+              fromBaseUnit(
+                firstStep.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+                tradeSellAsset.precision,
+              ),
+              tradeSellAsset.symbol,
+              {
+                maximumFractionDigits: 8,
+                omitDecimalTrailingZeros: true,
+                abbreviated: true,
+                truncateLargeNumbers: true,
+              },
+            ),
+            buyAmountAndSymbol: toCrypto(
+              fromBaseUnit(lastStep.buyAmountAfterFeesCryptoBaseUnit, tradeBuyAsset.precision),
+              tradeBuyAsset.symbol,
+              {
+                maximumFractionDigits: 8,
+                omitDecimalTrailingZeros: true,
+                abbreviated: true,
+                truncateLargeNumbers: true,
+              },
+            ),
+          }),
+          metadata: {
+            swapId: swap.id,
+          },
+          assetIds: [tradeSellAsset.assetId, tradeBuyAsset.assetId],
+          initiatorAccountId: sellAccountId,
+        }),
+      )
+    }
+  }, [
+    tradeExecution,
+    dispatch,
+    translate,
+    toCrypto,
+    tradeSellAsset,
+    tradeBuyAsset,
+    tradeQuote,
+    previousTradeExecutionState,
+    previousTradeQuoteId,
+    currentHopIndex,
+    previousTxHash,
+    sellAccountId,
+    previousQuoteOrRate,
+  ])
+
+  // Update swap with tx hash or swapId for chainflip when the user did the first step
+  useEffect(() => {
+    if (!tradeExecution) return
+    if (!tradeExecution.firstHop.swap.sellTxHash) return
+    if (!tradeQuote) return
+
+    const swap = selectSwapByQuoteId(store.getState(), {
+      quoteId: tradeQuote.id as unknown as QuoteId,
+    })
+
+    if (
+      tradeQuote.quoteOrRate === 'quote' &&
+      tradeExecution.state === TradeExecutionState.FirstHop &&
+      tradeExecution.firstHop.swap.sellTxHash &&
+      swap?.status === SwapStatus.Pending
+    ) {
+      dispatch(
+        swapSlice.actions.updateSwap({
+          id: swap.id,
+          sellTxHash: tradeExecution.firstHop.swap.sellTxHash,
+        }),
+      )
+    }
+  }, [tradeExecution, dispatch, tradeQuote, previousTxHash])
+
+  // Update actions status when swap is confirmed or failed
+  const actionsQueries = useMemo(() => {
+    return pendingSwapActions
+      .map(action => {
+        switch (action.type) {
+          case ActionCenterType.Swap: {
+            if (!isTradePayloadDiscriminator(action)) return undefined
+
+            const swap = swapByIds[action.metadata.swapId]
+
+            if (!swap) return undefined
+
+            return {
+              queryKey: ['actionCenterPolling', action.id, swap.id, swap.sellTxHash],
+              queryFn: () =>
+                getTradeStatusHandler({
```

Comment:
This looks off, it looks like a query, but it is not and does a *ton* of side effects.

This should probably return trade status only, and then have effects to react on whatever we need in this hook

Reply:
Though meh, potato tomato, it *may* be fine to do so in a single function even though that's a bit of a non-standard use case of react-query. 
With that being said, couldn't we declare the function in the same module so we don't have to dependency-inject toast and translate?

Reply:
Declared the function in the same module in [fbcd366](https://github.com/shapeshift/web/pull/9625/commits/fbcd366ded019ee7c8586022801e2a9625cc80dd)

This is nice!